src/services/rns/rns.processor.ts

```diff
@@ -64,13 +65,39 @@ async function registerTransfer (
   return transferDomain.id
 }
 
+// getDomainName: Decode domain name
+function getDomainName (decodedData: DecodedData, tokenId: string, batchprocess: string): string | undefined {
+  if (decodedData) {
+    let name: string | undefined
+
+    if (decodedData.params[0].value === batchprocess.toLowerCase()) {
+      // Batch registration
+      const rlpDecoded: RLPDecoded = RLP.decode(decodedData.params[2].value) as unknown as RLPDecoded
+      const domainNames = rlpDecoded[1].map(
+        (domainData: number[]) =>
+          Utils.hexToAscii(
+            Utils.bytesToHex(
+              domainData.slice(88, domainData.length)
+            )
+          )
+      )
+      name = domainNames.find((domain: string) => Utils.numberToHex(Utils.sha3(domain) as string) === tokenId)
+    } else {
+      // Single registration
+      name = Utils.hexToAscii('0x' + decodedData.params[2].value.slice(218, decodedData.params[2].value.length))
```

Comment:
this like could be cleaner if separated to new lines

Reply:
I did that in this new change. Looks much better assigning the `params` attribute to a shorter variable.